### PR TITLE
cxx-qt-gen: allow for optional #[qobject] macro

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
This allows for using CXX-Qt helpers with normal C++ classes, eg #[inherit] and #[cxx_override] etc.

### CODE REVIEW

File: crates/cxx-qt-gen/src/generator/cpp/constructor.rs [hunk 1]
```diff
@@ -20,6 +20,7 @@ fn default_constructor(
 ) -> GeneratedCppQObjectBlocks {
     GeneratedCppQObjectBlocks {
         methods: vec![CppFragment::Pair {
+            // TODO: if there is no qobject macro what do we do with the constructor?
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/constructor.rs:23]: TODO what should happen to constructor on non QObject tagged classes? do we still do the same ?

File: crates/cxx-qt-gen/src/generator/cpp/qobject.rs [hunk 2]
```diff
@@ -118,6 +121,8 @@ impl GeneratedCppQObject {
             .insert("#include <cxx-qt-common/cxxqt_maybelockguard.h>".to_owned());
 
         // Build the base class
+        //
+        // TODO: if there is no qobject macro should we assume the base?
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/qobject.rs:125]: TODO what should the base class be when there is no qobject macro ? Should it just be empty ?

File: crates/cxx-qt-gen/src/generator/cpp/constructor.rs [hunk 3]
```diff
@@ -35,7 +35,31 @@ fn default_constructor(
...
+            header: format!("explicit {class_name}();", class_name = qobject.ident),
+            source: formatdoc!(
+                r#"
+            {class_name}::{class_name}()
+              {base_class_line}
+              , ::rust::cxxqtlib1::CxxQtType<{rust_obj}>(::{namespace_internals}::createRs()){initializers}
+            {{ }}
+            "#,
+                base_class_line = if base_class.is_empty() {
+                    "".to_string()
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/constructor.rs:50]: needs a test

File: crates/cxx-qt-gen/src/generator/cpp/constructor.rs [hunk 4]
```diff
@@ -115,6 +139,7 @@ pub fn generate(
             header: format!(
                 "explicit {class_name}(::{namespace_internals}::CxxQtConstructorArguments{index}&& args);"
             ),
+            // TODO: what happens when base class is empty here?
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/constructor.rs:142]: what happens if we have a constructor but no base class?

File: crates/cxx-qt-gen/src/generator/cpp/inherit.rs [hunk 5]
```diff
@@ -23,6 +23,7 @@ pub fn generate(
 
     for method in inherited_methods {
         let return_type = syn_type_to_cpp_return_type(&method.method.sig.output, cxx_mappings)?;
+        // TODO: we need to error if there is no base class and no qobject macro ?
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/inherit.rs:26]: if inherit is used with no base class and no qobject macro, do we need to error?

File: crates/cxx-qt-gen/src/generator/cpp/constructor.rs (hunk 4 above)
Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/constructor.rs:141]: we always have a base class now

File: CHANGELOG.md [hunk 6]
```diff
@@ -32,6 +32,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `cxx-qt-gen` now does not generate code requiring `cxx-qt-lib`, this allows for `cxx-qt-lib` to be optional
 - `cxx-qt-lib` headers must be given to `cxx-qt-build` with `.with_opts(cxx_qt_lib_headers::build_opts())`
 - File name is used for CXX bridges rather than module name to match upstream
+- `#[qobject]` attribute is now options on types in `extern "RustQt"`
```
Comment (LeonMatthesKDAB) [inline CHANGELOG.md:35]:
```suggestion
- `#[qobject]` attribute is now optional on types in `extern "RustQt"`
```

**Review by LeonMatthesKDAB — COMMENTED** [review]
Small typo, otherwise looks good :+1:

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**